packages/components/src/form-toggle/test/index.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle, { noop } from '..';
+import type { FormToggleProps } from '../types';
+
+const getInput = () => screen.getByRole( 'checkbox' ) as HTMLInputElement;
+
+const ControlledFormToggle = ( { onChange }: FormToggleProps ) => {
+	const [ isChecked, setChecked ] = useState( false );
+	return (
+		<FormToggle
+			checked={ isChecked }
+			onChange={ ( value ) => {
+				setChecked( ( state ) => ! state );
+				onChange( value );
+			} }
+		/>
+	);
+};
+
+describe( 'FormToggle', () => {
+	describe( 'basic rendering', () => {
+		it( 'should render', () => {
+			render( <FormToggle onChange={ noop } /> );
+			expect( getInput() ).not.toBeNull();
+		} );
+
+		it( 'should render a span element with an unchecked checkbox', () => {
+			const { container } = render( <FormToggle onChange={ noop } /> );
+
+			expect( getInput() ).toHaveProperty( 'checked', false );
+			expect( container.firstChild ).toMatchSnapshot();
+		} );
+
+		it( 'should render a checked checkbox and change the accessibility text to On when providing checked prop', () => {
```

Comment:
```suggestion
		it( 'should render a checked checkbox when providing checked prop', () => {
```

Not sure what this part is. Maybe there was some a11y text in the past?

Reply:
Thanks. Updated 👍